lib/portlayer/exec/container.go

```diff
@@ -393,16 +393,24 @@ func infraContainers(ctx context.Context, sess *session.Session) ([]mo.VirtualMa
 
 // find the childVM for this resource pool by name
 func childVM(ctx context.Context, sess *session.Session, name string) (*vm.VirtualMachine, error) {
-	searchIndex := object.NewSearchIndex(sess.Client.Client)
-	child, err := searchIndex.FindChild(ctx, VCHConfig.ResourcePool.Reference(), name)
```

Comment:
I suggest using the container ID in the VmPathName, along the lines of:

```go
spec.Files = &types.VirtualMachineFileInfo{
		VmPathName: fmt.Sprintf("[%s] %s/%d.vmx", datastore, id, id),
}
```

Then here, you can use `SearchIndex.FindByDatastorePath` with the value of VmPathName.

The VM can have any name you choose, can be renamed, etc., and VmPathName will stay the same.